### PR TITLE
exoscale-cli: 1.12.0 -> 1.14.0

### PR DESCRIPTION
**Pull request by dramaturg**
###### Motivation for this change

Update to [latest release](https://github.com/exoscale/cli/releases/tag/v1.14.0) of the exoscale cli

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/admin/exoscale-cli/default.nix [hunk 1]
```diff
@@ -2,17 +2,17 @@
...
   src = fetchFromGitHub {
     owner  = "exoscale";
     repo   = "cli";
     rev    = "v${version}";
-    sha256 = "04ym7mfv565icj3lmd2nrvq9asawwmmzg09681pj9py61ws56bxr";
+    sha256 = "10sx542m6r6yq99nqn69bjfc6rv4nr9swi74rfccxb7dbifdrqpp";
   };
 
   goPackagePath = "github.com/exoscale/cli";
-  goDeps = ./deps.nix;
```
Comment (turion) [inline pkgs/tools/admin/exoscale-cli/default.nix:15]: How come this isn't needed anymore?

Comment (turion) [reply pkgs/tools/admin/exoscale-cli/default.nix:15]: I mean, is it a refactoring or is it necessary to make the new version build?

**Comment by turion** [comment]
Binary executes.

File: pkgs/tools/admin/exoscale-cli/default.nix (hunk 1 above)
Comment (turion) [reply pkgs/tools/admin/exoscale-cli/default.nix:15]: If it's a refactoring, you could put these lines in a separate commit such that the first commit only consists of bumping the version and the hash.

Comment (dramaturg) [reply pkgs/tools/admin/exoscale-cli/default.nix:15]: The upstream repo ships with all dependencies in `vendor/` which should have been used in the first nix package already, really.

**Comment by turion** [comment]
@timokau What am I doing wrong here?

**Comment by timokau** [comment]
The `needs_work` status doesn't exist anymore since https://github.com/timokau/marvin-mk2/issues/29. The closest equivalent is `awaiting_changes`. Most of the time you should just let Marvin do its thing though, you don't have to set the state explicitly. It sets to `awaiting_changes` when you leave a change-requesting review. Its has recently become a bit more conservative with switching to that state (https://github.com/timokau/marvin-mk2/pull/58), since it doesn't hurt much to have a PR in `awaiting_reviewer` by mistake but it may hurt a lot to have it in `awaiting_changes` by mistake.

Marvin should of course have linked the usage document on opt-in, it failed to do so (https://github.com/timokau/marvin-mk2/pull/62).

**Comment by turion** [comment]
@timokau Thanks for your detailed explanations.

/status awaiting_changes

File: pkgs/tools/admin/exoscale-cli/default.nix (hunk 1 above)
Comment (turion) [reply pkgs/tools/admin/exoscale-cli/default.nix:15]: @dramaturg Ok, makes sense, thanks! So it's a refactoring. I'd recommend putting it in a separate commit.

**Comment by dramaturg** [comment]
@turion  Err... what needs to be changed here?

**Comment by turion** [comment]
> @turion Err... what needs to be changed here?

As I said in https://github.com/NixOS/nixpkgs/pull/92705#discussion_r451515738 it would be nice to split into two commits where one contains your refactoring and the other does the version bump :)

File: pkgs/tools/admin/exoscale-cli/default.nix (hunk 1 above)
Comment (dramaturg) [reply pkgs/tools/admin/exoscale-cli/default.nix:15]: I refactored the changes into two separate commits.

**Comment by Mic92** [comment]
Result of `nixpkgs-review pr 92705` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
<br>- exoscale-cli
</details>

**Comment by turion** [comment]
Awesome!

/status needs_merger

**Pull request merged**